### DISCUSSION TITLE
Jenkins Cobertura Plugin vs. ReportGenerator

### DISCUSSION THREAD

**Discussion by jsk33** [danielpalme/ReportGenerator#432] [Q&A]
Hello :)

Any idea on why the coverage report numbers generated using the Jenkins Cobertura plugin may be different from the numbers generated by ReportGenerator using the same constituent xml files?
The command I'm running: `reportgenerator "-reports:<path>" "-targetdir:<path>" "-reporttypes:Cobertura"`

Thanks for your time!

**Comment by danielpalme** [comment]
This wiki page explains how ReportGenerator calculates its metrics: https://github.com/danielpalme/ReportGenerator/wiki/Calculation-details

I don't know how the Jenkins Cobertura plugin calculates its results.
If you can share a concrete example we might find the reason.

**Reply by jsk33** [reply]
Excuse the blurring of the picture, but these are some of the errors I see when I run ReportGenerator on my folder of Cobertura coverage xml files:
![screenshot](https://user-images.githubusercontent.com/36433643/122293508-14eeb700-cec5-11eb-9caa-7ad26f6ae848.png)

It still generates the merged xml report in the end, but the numbers are slightly different compared to what I see from Jenkins (around 3% off), which I need to use as the "source of truth". 

Are you familiar with the errors in the screenshot and whether they would be affecting the merging process?

**Reply by danielpalme** [reply]
The merging should not be affected by the error messages.
The messages indicate that some files don't exist at the expected location any more.

Without a concrete Cobertura file and one or two sample classes it's hard to determine if there is a problem.
